Beta v6.20.2 => v6.20.3

Fourdee opened the pull request:
### Changelog between beta versions v6.20.2 and v6.20.3

#### Changes / Improvements / Optimisations
- DietPi-Software | Shairport Sync: Updated to 3.2.2 (previously 3.1.7): https://github.com/Fourdee/DietPi/issues/2439
- DietPi-Software | Gitea: New installations now install 1.7 (previously 1.4). Many thanks to @yumiris for this update! https://github.com/Fourdee/DietPi/pull/2448
- DietPi-Software | Grafana: Now the new official APT repo is used for all architectures (excluding ARMv6), which allows updates to the current upstream version (currently v5.4.3) as well. This change will be applied to existing installs on dietpi-update to v6.20. Thanks to @jvteleco for informing us: https://github.com/Fourdee/DietPi/issues/2449

#### Bug Fixes
- DietPi-Software | PHP: Resolved an issue where install fails on Stretch due to package name changes in the Debian repo. Thanks to @rox-wolf for reporting this issue: https://github.com/Fourdee/DietPi/issues/2438
- DietPi-LetsEncrypt | Resolved an issue where Nginx and Apache modules are missing, if one installs CertBot first and the webserver afterwards on a separate dietpi-software run: https://github.com/Fourdee/DietPi/issues/2295

MichaIng committed:
v6.20

+ DietPi-Config | Hide Audio and Performance Options on VM
+ DietPi-Config | RPi: Check config.txt options for *disable*=1 options, instead of *disable*=0, where default values are 0, otherwise when setting is commented or removed by user, estimation is wrong.
+ DietPi-Config | RPi: Use vcgencmd to estimate config.txt settings, where default values depend on device and/or other settings (HDMI boost, overscan)
+ DietPi-Config | RPi: Reorder display options code to each current value estimation beside related menu entry 
+ DietPi-Config | RPi: Estimate headless mode correctly via hdmi_ignore_hotplug=1 && hdmi_ignore_composite=1 settings
+ DietPi-Config | Minor coding

MichaIng committed:
v6.20

+ DietPi-Set_Hardware | Headless mode: Apply Odroid settings via G_CONFIG_INJECT
+ DietPi-Set_Hardware | Headless mode: When disabling headless mode on RPi, leave boot rainbow splash disabled, which was as well default before implementing new config.txt-based headless mode

MichaIng committed:
v6.20

+ RPi | config.txt: Add "hdmi_blanking" setting and info, since we set this with headless mode now
+ RPi | config.txt: Add info about "enable_uart=1" forcing "core_freq=250"

MichaIng committed:
v6.20

+ DietPi-Config | Minor syntax fix

MichaIng committed:
v6.20

+ DietPi-Patch | "s/^CONFIG_HDMI_OUTPUT=/AUTO_SETUP_HEADLESS=/" lead to doubled entry, since AUTO_SETUP_HEADLESS was already added by "verify_dietpi.txt". Remove old setting only and remove doubled entry from Beta v6.20.2 testers, which otherwise causes G_CONFIG_INJECT to fail.

MichaIng committed:
v6.20

+ DietPi-PreBoot | Enable first run headless mode for Odroids C1 and C2 as well

MichaIng committed:
v6.20

+ DietPi-Config | RPi: Fix headless mode estimation on PSU noise reduction option as well
+ DietPi-Config | Minor coding: G_CONFIG_INJECT implementation and [On]/[Off] option consistency

Fourdee committed:
Merge pull request #2437 from Fourdee/dietpi-config

DietPi-Config/Set_Hardware | Headless mode fine tuning and coding

Commit:
v6.20

+ Revert to config.txt values, so that changes made in GUI will be reflected: https://github.com/Fourdee/DietPi/pull/2437#issuecomment-457633681

MichaIng committed:
v6.20

DietPi-Software | Nextcloud Talk: Remove `lt-cred-mech` from config, which is not required/conflicting with `use-auth-secret`
- Related log entries:
```
Jan 19 23:08:20 my.domain.org turnserver[6712]: CONFIGURATION ALERT: You specified --lt-cred-mech and --use-auth-secret in the same time.
Jan 19 23:08:20 my.domain.org turnserver[6712]: Be aware that you could not mix the username/password and the shared secret based auth methohds.
Jan 19 23:08:20 my.domain.org turnserver[6712]: Shared secret overrides username/password based auth method. Check your configuration!
```
- Related config file comments:
```
# Be aware that use-auth-secret overrides some part of lt-cred-mech.
# Notice that this feature depends internally on lt-cred-mech, so if you set
# use-auth-secret then it enables internally automatically lt-cred-mech option
# like if you enable both.
#
# You can use only one of the to auth mechanisms in the same time because,
# both mechanism use the username and password validation in different way.
#
# This way be aware that you can't use both auth mechnaism in the same time!
# Use in config either the lt-cred-mech or the use-auth-secret
# to avoid any confusion.
```
DietPi-Software | Nextcloud Talk: Port must be positive; Minor coding

MichaIng committed:
v6.20

+ DietPi-Software | PHP: On Stretch, install php7.0-* APT packages instead of meta php-* packages, which are not available anymore at least in case of php-opcache
+ DietPi-Software | PHP: php-redis (same as php-apcu) has no php7.X-redis package, use $PHP_BINARY-redis, which also fits php5-redis on Jessie
+ DietPi-Software | PHP: The above allows minor simplicity in PHP-FPM socket path estimation

MichaIng committed:
v6.20

+ CHANGELOG | PHP: Resolved an issue where install fails on Stretch due to package name changes in the Debian repo.

Commit:
v6.20

+ Switch to ARMbian based images.

Commit:
Merge branch 'dev' of https://github.com/Fourdee/DietPi into dev

Commit:
v6.20

+ Fix:
 cannot access '/etc/profile.d/dietpi-*.sh': No such file or directory

Commit:
v6.20

+ Fix kill-ssh-user-sessions-before-network, no longer used.

Commit:
v6.20

+ Rock64 WG
- RockPro64

Commit:
v6.20

+ Correction.

Commit:
v6.20

+ Rock64 WG testing
+ Fix server/client choice "NULL"

Commit:
v6.20

+ Revert rock64 enable: https://github.com/Fourdee/DietPi/issues/2052#issuecomment-457815683

Commit:
v6.20

- DietPi-Software | Shairport Sync: Updated to 3.2.2 (previously 3.1.7): https://github.com/Fourdee/DietPi/issues/2439

+ Jessie removal.

Fourdee committed:
Merge pull request #2441 from Fourdee/php

DietPi-Software | PHP: Fix PHP package naming due to Debian repo changes in Stretch

Commit:
v6.20

+ Fix search.

Commit:
Merge branch 'dev' of https://github.com/Fourdee/DietPi into dev

Commit:
v6.20

+ RC beta up

MichaIng committed:
v6.20

+ CHANGELOG | DietPi-LetsEncrypt: Resolved an issue where Nginx and Apache modules are missing, if one installs CertBot first and the webserver afterwards

MichaIng commented:
@Fourdee 
Before you merge, I have one tiny enhancement about `config.txt` vs `vcgenmod` detection: Check `config.txt` first and only use `vcgenmod` if no active setting was detected, thus a depending default is active. Will open PR.

**EDIT**: Done via commit, benefits from both worlds: https://github.com/Fourdee/DietPi/commit/6748a34cc8b9c25893ae1e6cd129126ff2a1fd70

MichaIng committed:
v6.20

+ DietPi-Config | RPi: Attempt to find "config.txt" entry first, otherwise use "vcgencmd" to estimate currently active value

MichaIng commented on PREP_SYSTEM_FOR_DIETPI.sh:
```diff
@@ -548,12 +548,11 @@
 		rm "DietPi-$GITBRANCH/dietpi/pre-patch_file"
 		rm "DietPi-$GITBRANCH/dietpi/patch_file"
 
+		l_message='Set execute permissions for DietPi scripts' G_RUN_CMD chmod +x -R "DietPi-$GITBRANCH"
```
@Fourdee 
Little complain: We will set execute permissions to settings files that definitely don't need them. Also the previous solution below was failsafe by (re)setting execute permissions as well to files that might not be included in the update.

/etc/profile.d/dietpi-*.sh /etc/bashrc.d/dietpi-*.sh actually not require execute permissions, since they are sourced.

**EDIT: Ah jep `/etc/profile.d/dietpi-*.sh` is not used anymore at all. All our scripts are not in bashrc.d which makes sense, since all require bash.
DietPi-Update was already patched accordingly, seems I forgot to do the same with DietPi-PREP.**

MichaIng commented on dietpi/dietpi-software:
```diff
@@ -15272,7 +15222,7 @@ _EOF_
 				do
 
-					if (( ${aSOFTWARE_AVAIL_G_HW_MODEL[$i,$G_HW_MODEL]:=1} && ${aSOFTWARE_AVAIL_G_HW_ARCH[$i,$G_HW_ARCH]:=1} && ${aSOFTWARE_AVAIL_G_DISTRO[$i,$G_DISTRO]:=1} )) &&
+					if (( ${aSOFTWARE_AVAIL_G_HW_MODEL[$i,$G_HW_MODEL]:-1} && ${aSOFTWARE_AVAIL_G_HW_ARCH[$i,$G_HW_ARCH]:-1} && ${aSOFTWARE_AVAIL_G_DISTRO[$i,$G_DISTRO]:-1} )) &&
```
@Fourdee 
I used assigning here for the reason that those array entries are not pre-defined, but used afterwards. I think empty entries are handled later as well, when it comes to an install, but using `:=1` avoids the need to re-set/assign the value again every time it is checked.

Fourdee replied:
@MichaIng 

> but using :=1 avoids the need to re-set/assign the value again every time it is checked.

Ah apologies, in my testing, ```:-1``` and ```:=1``` have the same effect, both do the same task of static assign?

MichaIng replied:
@Fourdee 
`${var:-1}` will insert `1` only to the current statement, AFAIK, while
`${var:=1}` will assign `1` to the variable as well, so the next time `$var` is called it already has the value `1`.

Practically in the script currently it should not make a difference, but in general here we should assign `1` as default value for whatever step checks it next.
Of course in case of `var=${var:-1}` it does not make any different and using `:=1` would be a doubled assignment.
_______
Tiny verification, jep that's it:
```
root@VM-Jessie:~# echo $test

root@VM-Jessie:~# echo ${test:-new}
new
root@VM-Jessie:~# echo $test

root@VM-Jessie:~# echo ${test:=new}
new
root@VM-Jessie:~# echo $test
new
```
_________
I reverted the change, since we indeed want assigned values here.

MichaIng committed:
v6.20

+ DietPi-Patch | Remove doubled Shairport Sync reinstall
+ DietPi-Patch | Minor coding: Remove double quotes when assigning single variables/command expansions to a variable, use single quotes when porting "dietpi.txt" WiFi settings to  "dietpi-wifi.db"

MichaIng committed:
v6.20

+ DietPi-PREP | Add enable new/renamed "dietpi-kill_ssh"
+ DietPi-PREP | Tiny coding

MirisWisdom committed:
Update Gitea to 1.7

MichaIng committed:
v6.20

+ DietPi-Software | Grafana: Switch to official APT repo witch all archs

MichaIng committed:
v6.20

+ DietPi-Software | Grafana: All archs now use (and are patched to) official APT repo, thus x86 specific "grafana_stable.list" does not exist anymore

MichaIng committed:
v6.20

+ DietPi-Patch | Grafana: Remove obsolete x86_64 APT source, reinstall to switch all systems to the new official repo

MichaIng committed:
v6.20

+ CHANGELOG | Grafana: Now the new official APT repo is used for all architectures

Fourdee committed:
Merge pull request #2448 from yumiris/dev

DietPi-Software | Gitea 1.4 -> 1.7

Fourdee committed:
Merge pull request #2450 from Fourdee/grafana

DietPi-Software | Grafana: Switch to official APT repo witch all archs

Commit:
v6.20

- DietPi-Software | Gitea: New installations now install 1.7 (previously 1.4). Many thanks to @yumiris for this update! https://github.com/Fourdee/DietPi/pull/2448

Commit:
v6.20

+ Conflicts

Commit:
v6.20

+ Remove armbian banner/profiles left on ASUS TB image

Also possible kernel/APT updates reapply these files,  will see.

MichaIng committed:
v6.20

+ DietPi-Patch | Tiny: "rm -f" preferable over "&> /dev/null" (when using wildcard) to suppress missing file errors but print e.g. file system corruption and other issues that we might want to know.

Commit:
v6.20

+ Symlink permissions also required.

Commit:
Merge branch 'dev' of https://github.com/Fourdee/DietPi into dev

Commit:
v6.20

+ Grafana port change.

Commit:
v6.20

+ Add previous support for ARMv6 install: https://github.com/Fourdee/DietPi/pull/2450

MichaIng committed:
v6.20

+ DietPi-Update | Apply execution permissions to /DietPi/dietpi subdir only, since config.txt/dietpi.txt/boot.ini and such do not require them.
+ DietPi-Update | Minor coding: Double quotes not required when assigning single variable/command expansion to another variable, but should be used when passing a variable to G_CHECK_VALIDINT to avoid word splitting: 2. word = min, 3. word = max value

MichaIng committed:
v6.20

+ DietPi-PREP | Set execute permissions only to DietPi core scripts and cron jobs, nothing else requires them (bashrc.d as well not, since those are sourced)

MichaIng committed:
v6.20

+ DietPi-Update | "/etc/bashrc.d/dietpi-*.sh" do not require execute permissions, since those are sourced and must never be executed directly

MichaIng committed:
v6.20

+ DietPi-Software | Revert ":-1" => ":=1" since we want assigned values for future checks

MichaIng committed:
v6.20

+ DietPi-Config | RPi consistency: Check for "config.txt" entry first, use "vcgenmod" otherwise;
+ DietPi-Config | Coding: "framebuffer_x=${framebuffer_x#*=}" is muuuch faster then "sed 's/^[^=]*=//'", but more code and less known. However lets stay with performance here, although in most other cases we use "sed".

MichaIng committed:
v6.20

+ DietPi-Config | RPi: Framebuffer estimation second occurrence

MichaIng committed:
v6.20

+ DietPi-Config | RPi: Do not scrape commented "config.txt" values; Minor coding

Fourdee commented:
Ok merged. 

Ideally, v6.20 release tomorrow+. Pending day of testing.
